htmresearch/algorithms/column_pooler.py

```diff
@@ -273,25 +280,68 @@ def _computeInferenceMode(self, feedforwardInput, lateralInputs):
         lateralInput, self.connectedPermanenceDistal)
       numActiveSegmentsByCell[overlaps >= self.activationThresholdDistal] += 1
 
-    # Activate some of the feedforward supported cells
-    minNumActiveCells = self.sdrSize / 2
-    chosenCells = self._chooseCells(feedforwardSupportedCells,
-                                    minNumActiveCells, numActiveSegmentsByCell)
 
-    # If necessary, activate some of the previously active cells
-    if len(chosenCells) < minNumActiveCells:
-      remaining = numpy.setdiff1d(prevActiveCells, feedforwardSupportedCells)
-      remaining = remaining[numActiveSegmentsByCell[remaining] > 0]
+    chosenCells = []
+    minNumActiveCells =  30 #self.sdrSize / 2
+
+
+    # # This is the old method. It relies heavily on the chooseCells function, which is
+    # # a bit complex. It works well, but fares badly in the face of random-SDR insertions
+    # # in input sequences!
+    # chosenCells = self._chooseCells(feedforwardSupportedCells,
+    #                                 minNumActiveCells, numActiveSegmentsByCell)
+    #
+    # # If necessary, activate some of the previously active cells
+    # if len(chosenCells) < minNumActiveCells:
+    #   remaining = numpy.setdiff1d(prevActiveCells, feedforwardSupportedCells)
+    #   remaining = remaining[numActiveSegmentsByCell[remaining] > 0]
+    #
+    #   chosenCells = numpy.append(
+    #     chosenCells, self._chooseCells(remaining,
+    #                                    minNumActiveCells - len(chosenCells),
+    #                                    numActiveSegmentsByCell))
+
+
+    # New method. Doesn't use the chooseCells function.
+    # # First, select FF-supported cells. But if some have active segments, only chose these ones (they inhibit everyone else)
+    remainingFFSupportedCells = numpy.setdiff1d(feedforwardSupportedCells, chosenCells)
+    activeSegs = numActiveSegmentsByCell[remainingFFSupportedCells]
```

Comment:
I think this variable name should indicate that it's the "# of segments for feedforward supported cells".  Perhaps `numActiveSegsForFFSCells`.

It corresponds to the `remainingFFSupportedCells` was used to create it, so it's useful if their variable names indicate that.